src/haddock/modules/analysis/contactmap/contmap.py

```diff
@@ -0,0 +1,1843 @@
+
+        return res_res_contacts
+
+
+class ClusteredContactMap():
+    """ContactMap analysis for set of clustered structures."""
+
+    def __init__(
+            self,
+            models: list[Path],
+            output: Path,
+            params: dict,
+            ) -> None:
+        self.models = models
+        self.output = output
+        self.params = params
+        self.terminated = False
+
+    def run(self):
+        """Process analysis of contacts of a set of PDB structures."""
+        # initiate holding variables
+        clusters_contacts = {}
+        keys_list = []
+        # loop over models/structures
+        for pdb_path in self.models:
+            # initiate object
+            contact_map_obj = ContactsMap(
+                pdb_path,
+                f'{self.output}_{pdb_path.stem}',
+                self.params,
+                )
+            # Run it
+            pdb_contacts = contact_map_obj.run()
+            # Parse outputs
+            for cont in pdb_contacts:
+                # Check key
+                combined_key = f'{cont["res2"]}/{cont["res1"]}'
+                if combined_key not in clusters_contacts.keys():
+                    combined_key = f'{cont["res1"]}/{cont["res2"]}'
+                    if combined_key not in clusters_contacts.keys():
+                        keys_list.append(combined_key)
+                        clusters_contacts[combined_key] = {
+                            k: []
+                            for k in cont.keys()
+                            if k not in ["res1", "res2"]
+                            }
+                # Add data
+                for dtk in clusters_contacts[combined_key].keys():
+                    clusters_contacts[combined_key][dtk].append(cont[dtk])
+
+        combined_clusters_list = []
+        # Loop over keys
+        for combined_key in keys_list:
+            # point data
+            res1, res2 = combined_key.split('/')
+            dt = clusters_contacts[combined_key]
+
+            # Compute averages and ratio
+            avg_ca_ca_dist = np.mean(dt['ca-ca-dist'])
+            ca_ca_above_thresh = [
+                v for v in dt['ca-ca-dist']
+                if v <= self.params['ca_ca_dist_threshold']
+                ]
+            ca_ca_cont_ratio = len(ca_ca_above_thresh) / len(dt['ca-ca-dist'])
+
+            avg_shortest = np.mean(dt['shortest-dist'])
+            short_ab_t = [
+                v for v in dt['shortest-dist']
+                if v <= self.params['shortest_dist_threshold']
+                ]
+            shortest_cont_ratio = len(short_ab_t) / len(dt['shortest-dist'])
+
+            # most representative contact type
+            cont_ts = list(set(dt['contact-type']))
+            cont_t = sorted(
+                cont_ts,
+                key=lambda k: cont_ts.count(k),
+                reverse=True,
+                )[0]
+            cont_t_ratio = cont_ts.count(cont_t) / len(dt['contact-type'])
```

Comment:
If I understand correctly this part of the code loops over the clusters and the keys to extract some statistics such as the shortest_cont_ratio ..can you add some comments here and there to clarify what these quantities are?